src/Hypsography/Hypsography.jl

```diff
@@ -156,7 +160,46 @@ function ExtrudedFiniteDifferenceSpace(
     )
 end
 
-
+"""
+  diffuse_surface_elevation!(f; κ,iter,dt)
+Option for 2nd order diffusive smoothing of generated terrain.
+Compute smoothed profile before assigning the surface
+elevation to the `HypsographyAdaption` type. Laplacian 
+operator is applied with simple Euler updates to generate
+profiles for each new iteration.
+(1) Compute discrete elevation profile f
+(2) Compute diffuse_surface_elevation!(f, κ, iter). f is mutated.
+(3) Define `Hypsography.LinearAdaption(f)`
+(4) Define `ExtrudedFiniteDifferenceSpace` with new surface elevation.
+Default diffusion parameters are appropriate for spherical arrangements. 
+For `zmax-zsfc` == 𝒪(10^4), κ == 𝒪(10^8), dt == 𝒪(10⁻¹).
+Applicable on periodic domains only. 
+"""
+function diffuse_surface_elevation!(
+    f::Fields.Field;
+    κ::T = 1e8,
+    maxiter::Int = 100,
+    dt::T = 1e-1,
+) where {T}
+    # Define required ops
+    wdiv = Operators.WeakDivergence()
+    grad = Operators.Gradient()
+    FT = eltype(f)
+    # Create dss buffer
+    ghost_buffer = (bf = Spaces.create_dss_buffer(f),)
+    # Apply smoothing
+    for iter in 1:maxiter
+        # Euler steps
+        χf = @. wdiv(grad(f))
+        Spaces.weighted_dss_start!(χf, ghost_buffer.bf)
+        Spaces.weighted_dss_internal!(χf, ghost_buffer.bf)
+        Spaces.weighted_dss_ghost!(χf, ghost_buffer.bf)
+        @. f += κ * dt * χf
```

Comment:
Should we use `ClimaTimeSteppers` here?

Reply:
@akshaysridhar would there be any advantage to using a different timestepper here? If so, we can add it in.

Reply:
I see DiffEqBase is in the deps here; this is supposed to be a relatively light component in the overall scheme - a Forward-Euler update will suffice (hence the current form). Updating it to use DiffEqBase with new commits to follow soon.